### PR TITLE
ZER-49 Return results of calculation in compute API

### PR DESCRIPTION
**Pull request by ochupa**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/serializers/field_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    if object.name.present?
+      object.name.parameterize.underscore
+    else
+      object.name
+    end
+  end
+
+  def result
+    object.result
+  end
```
Comment (maksymst) [inline app/serializers/field_serializer.rb:16]: no need to defined this method. It should implicitly delegate `result` to `object`

File: app/serializers/field_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class FieldSerializer < ActiveModel::Serializer
+  attributes :name, :result
+
+  def name
+    if object.name.present?
+      object.name.parameterize.underscore
+    else
+      object.name
+    end
```
Comment (maksymst) [inline app/serializers/field_serializer.rb:11]: what the purpose of this block?

File: spec/requests/calculators_spec.rb [hunk 3]
```diff
@@ -4,10 +4,38 @@
...
+      json_response = JSON.parse(response.body)
+      expect(json_response['result'][0].keys).to contain_exactly(
+        'name',
+        'result'
+      )
+    end
+
+    it 'JSON response contains field `name` in snake case format' do
+      json_response = JSON.parse(response.body)
+      # debugger
```
Comment (maksymst) [inline spec/requests/calculators_spec.rb:37]: need this?

File: spec/requests/calculators_spec.rb [hunk 4]
```diff
@@ -4,10 +4,38 @@
...
+    before do
       post compute_api_v1_calculator_path(calculator)
+    end
+
+    it 'returns JSON' do
       expect(response).to have_http_status(200)
     end
+
+    it 'JSON response contains `result` in the root' do
+      json_response = JSON.parse(response.body)
```
Comment (maksymst) [inline spec/requests/calculators_spec.rb:23]: `JSON.parse(response.body)` is used in every method

**Review by maksymst — CHANGES_REQUESTED** [review]

File: spec/requests/calculators_spec.rb (hunk 3 above)
Comment (ochupa) [reply spec/requests/calculators_spec.rb]: :-) 
Removed.

File: app/serializers/field_serializer.rb (hunk 2 above)
Comment (ochupa) [reply app/serializers/field_serializer.rb]: Thanks. I've simplified this method

File: app/serializers/field_serializer.rb (hunk 1 above)
Comment (ochupa) [reply app/serializers/field_serializer.rb]: Thank you for explanation.

**Review by maksymst — APPROVED** [review]
Good job 👍

**Review by haliapats — APPROVED** [review]

**Pull request merged**